mountpoint-s3-client/tests/auth.rs

```diff
@@ -187,12 +187,96 @@ async fn test_profile_provider_async() {
     let _result = S3CrtClient::new(config).expect_err("profile doesn't exist");
 }
 
+async fn test_profile_provider_assume_role_async() {
+    let sdk_client = get_test_sdk_client().await;
+    let subsession_role = get_subsession_iam_role();
+    let (bucket, prefix) = get_test_bucket_and_prefix("test_profile_provider_assume_role");
+
+    let key = format!("{prefix}/hello");
+    let body = b"hello world!";
+    sdk_client
+        .put_object()
+        .bucket(&bucket)
+        .key(&key)
+        .body(ByteStream::from(Bytes::from_static(body)))
+        .send()
+        .await
+        .unwrap();
+
+    let profile_name = "mountpoint-profile";
+    let source_profile = "default";
+    let mut config_file = NamedTempFile::new().unwrap();
+    writeln!(config_file, "[profile {}]", profile_name).unwrap();
+
+    // Set up the environment variables to use this new config file. This is only OK to do because
+    // this test is run in a forked process, so won't affect any other concurrently running tests.
+    std::env::set_var("AWS_CONFIG_FILE", config_file.path().as_os_str());
+
+    // Credentials are provided as environment variables when running on GitHub Actions,
+    // so we will have to build a credentials file from those variables if they exist, otherwise
+    // we are going to use default profile from the test instance.
+    let _credentials_file = get_credentials_from_env().map(|credentials| {
+        let mut credentials_file = NamedTempFile::new().unwrap();
+        writeln!(credentials_file, "[{}]", source_profile).unwrap();
+        writeln!(credentials_file, "aws_access_key_id={}", credentials.access_key_id()).unwrap();
+        writeln!(
+            credentials_file,
+            "aws_secret_access_key={}",
+            credentials.secret_access_key()
+        )
+        .unwrap();
+        if let Some(session_token) = credentials.session_token() {
+            writeln!(credentials_file, "aws_session_token={session_token}").unwrap();
+        }
+        std::env::remove_var("AWS_ACCESS_KEY_ID");
+        std::env::remove_var("AWS_SECRET_ACCESS_KEY");
+        std::env::remove_var("AWS_SESSION_TOKEN");
+        std::env::set_var("AWS_SHARED_CREDENTIALS_FILE", credentials_file.path().as_os_str());
+        credentials_file
+    });
```

Comment:
Can we adopt the same logic as the fork tests, using the default creds provider to always create a config file?